### DISCUSSION TITLE
What's the correct way to use setif()?

### DISCUSSION THREAD

**Discussion by GalinhaLX** [mavoweb/mavo#920] [Q&A] [answered]
Hi,

On this [codepen](https://codepen.io/galinhalx/pen/WNJxyvJ)
I'm trying to use the shortcut of 

 `<button mv-action="if(quantity > 0, set(quantity, quantity - 1))">removes 1</button>`

with setif()
  
`  <button mv-action="setif(quantity > 0, quantity - 1)">Should remove 1 if quantity is bigger than 0 </button>`

which doesn't work

what's the correct way to tell mavo, which property it should apply the new value?

**Comment by DmitrySharabin** [comment, marked as answer]
Hey there,

The correct form of the `setif()` action is: `setif(condition, property, expression)`. So, you just add the new (third) `condition` argument  _before_ the exsisting two arguments of the `set()` action (and rename the action, of course 😊).

In your case, it would look like this: `setif(quantity > 0, quantity, quantity - 1)`.
